Final piece of the toolchain upgrade (gcc6.x)

jbech-linaro opened the pull request:
This is the patch that step up default toolchain in build.git to GCC6.x. Note that I had to add a dummy patch for Travis so that it was actually checking out GCC6.x when running the Travis job. That should not be included when merging this. I'll remove that when adding R-B tags etc. However, my push looked good on my local branch, see -> https://travis-ci.org/jbech-linaro/build/builds/199189080                                                   
                                                                                                                
So, hopefully this should be it.

jenswi-linaro commented:
`Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>`

jforissier commented:
Nitpicking: you could as well mention the actual version in the commit description, i.e., GCC 6.2 rather than 6.x. Anyway, that's good!
`Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>`

jbech-linaro commented:
I'll change to the correct version number as proposed when rebasing and adding RB.

jbech-linaro commented:
Travis was happy with the first patch set, so now I've added the tags, removed the "temporary" commit, changed to GCC6.2.

Squashed, rebased and tag(s) applied (should be) ready for merge! Thanks!
